agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -78,11 +109,11 @@ object AtumContext {
      *  @return
      */
     def createCheckpoint(checkpointName: String)(implicit atumContext: AtumContext): DataFrame = {
-      atumContext.measurements.foreach { measure =>
+      atumContext.measures.foreach { measure =>
         val result = MeasureResult(measure, measure.function(df))
         AtumAgent.publish(checkpointName, atumContext, result)
 
-        executeMeasures(checkpointName, atumContext.measurements)
+        executeMeasures(checkpointName, atumContext.measures)
```

Comment:
this line can be removed, it's doing the same as the whole foreach basically

Reply:
Will removed, but as said, this needs reword. `createCheckpoint` and send it as an "atomic" steo.